app/models/pokemon.js

```diff
@@ -34,8 +34,45 @@ var pokemonGoBasicSchema = new Schema({
         femaleRatio: Number,
         breedable: Boolean
     }
-});
+};
+
+var pokemonGoBasicSchema = new Schema(pokemon);
 
 pokemonGoBasicSchema.index({"pokemonId": 1}, {"unique": true});
 
-module.exports = mongoose.model('pokemon', pokemonGoBasicSchema);
+module.exports = {
+
+    getModel: function () {
+        return pokemon;
+    },
+    getSchema: function () {
+        return mongoose.model('pokemon', pokemonGoBasicSchema);
+    },
+    getMappedModel: function (pokemon) {
+        return {
+            pokemonId: pokemon.id || pokemon.pokemonId,
+            name: pokemon.name,
+            icon: pokemon.icon,
+            classification: pokemon.classification,
+            types: pokemon.type && pokemon.type.toLowerCase(),
+            resistance: pokemon.resistance && pokemon.resistance.toLowerCase(),
+            weakness: pokemon.weakness && pokemon.weakness.toLowerCase(),
+            // fastAttacks: [{name: {type: String}, type: {type: String}, damage: {type: Number}}],
```

Comment:
Its not yet clear about how the query parameters should be for these features.. hence I have left them commented for now.